hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1423,7 +1422,7 @@ aggregate st = \case
             { coordinatedHeadState =
                 coordinatedHeadState
                   { localUTxO = newLocalUTxO
-                  , pendingDeposits = Map.delete recoveredTxId existingDeposits
+                  , pendingDeposits = filter (\(depositTxId, _) -> depositTxId /= recoveredTxId) existingDeposits
```

Comment:
ah but UTxO.delete would be different since we are not dealing with `UTxO` here. existingDeposits were a `Map TxId (a,b,c)` and now are a `[(a,b)]` where a and b are not `(TxIn, TxOut)`.